### PR TITLE
PP-8804 Stripe setup - Update director controller for POST

### PR DESCRIPTION
**Pull request by kbottla**
## WHAT
- Updated director controller for POST
  - POST is context aware (i.e., new services onboarding, kyc, switch PSP)
- Will raise separate PR for cypress tests

### CODE REVIEW

File: app/controllers/stripe-setup/director/post.controller.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+      dobMonth: formFields[DOB_MONTH_FIELD],
+      dobYear: formFields[DOB_YEAR_FIELD]
+    }
+
+    pageData['errors'] = errors
+
+    return response(req, res, 'stripe-setup/director/index', {
+      ...pageData, isSwitchingCredentials, collectingAdditionalKycData, currentCredential
+    })
+  } else {
```
Comment (stephencdaly) [inline app/controllers/stripe-setup/director/post.controller.js:74]: As we return if there are errors we could remove the `else` to reduce nesting. Not something that needs to be fixed, just a personal preference for readability

File: app/controllers/stripe-setup/director/post.controller.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+      } else {
+        return res.redirect(303, formatAccountPathsFor(paths.account.stripe.addPspAccountDetails, req.account && req.account.external_id))
+      }
+    } catch (err) {
+      next(err)
+    }
+  }
+}
+
+async function getStripeAccountId (account, isSwitchingCredentials, correlationId) {
```
Comment (stephencdaly) [inline app/controllers/stripe-setup/director/post.controller.js:109]: Could this be shared as a utility functions, as I'm sure we do this in a lot of places?

**Review by stephencdaly — APPROVED** [review]
Looks good to me.

File: app/controllers/stripe-setup/director/post.controller.js (hunk 2 above)
Comment (kbottla) [reply app/controllers/stripe-setup/director/post.controller.js:109]: yes, going to introduce utility and refactor separately.

**Pull request merged**